[WebCore] Remove tertiarySystemFillColor selector check

LumioseSil opened the pull request:
#### 84c6a1eb31e3d5f9e8163c2b420664ba413d7ba4
<pre>
[WebCore] Remove tertiarySystemFillColor selector check
<a href="https://bugs.webkit.org/show_bug.cgi?id=278999">https://bugs.webkit.org/show_bug.cgi?id=278999</a>

Reviewed by Aditya Keerthi.

HAVE(NSCOLOR_FILL_COLOR_HIERARCHY) already ensures that this code can
only run on AppKit with tertiarySystemFillColor, meaning that we can
remove the respondsToSelector check.

See:
<a href="https://developer.apple.com/documentation/appkit/nscolor/4236446-tertiarysystemfillcolor?language=objc">https://developer.apple.com/documentation/appkit/nscolor/4236446-tertiarysystemfillcolor?language=objc</a>

* Source/WebCore/rendering/mac/RenderThemeMac.mm:
  (WebCore::RenderThemeMac::systemColor const): Remove
  selector check.

Canonical link: <a href="https://commits.webkit.org/283115@main">https://commits.webkit.org/283115@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/282543fee4cfcf1cb3b0b157a0a7a47b980e3114

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65292 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/44661 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17906 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69316 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/15898 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/67410 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/52443 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16180 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/52441 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/15898 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68358 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/41264 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/56510 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/33065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37938 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/13882 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14775 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/59804 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14222 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/71021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9244 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/13687 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/71021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9276 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56572 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/71021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14385 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7632 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/1295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/40471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/41548 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/42729 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41292 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (changes requested):
It's fine to remove the selector check, since `HAVE(NSCOLOR_FILL_COLOR_HIERARCHY)` will protect us on older OSes.

But the commit message is wrong. The documentation link is incorrect, since the link is for UIKit. The code is here private AppKit API that was only added in macOS 14 (which is the other incorrect bit, as the commit message references "macOS 13").

Requesting changes for the commit message, I will approve once that is fixed.

LumioseSil committed:
[WebCore] Remove tertiarySystemFillColor selector check
https://bugs.webkit.org/show_bug.cgi?id=278999

Reviewed by Aditya Keerthi.

HAVE(NSCOLOR_FILL_COLOR_HIERARCHY) already ensures that this code can
only run on AppKit with tertiarySystemFillColor, meaning that we can
remove the respondsToSelector check.

See:
https://developer.apple.com/documentation/appkit/nscolor/4236446-tertiarysystemfillcolor?language=objc

* Source/WebCore/rendering/mac/RenderThemeMac.mm:
  (WebCore::RenderThemeMac::systemColor const): Remove
  selector check.

Canonical link: https://commits.webkit.org/283115@main
